pkg/syncer/syncer.go

```diff
@@ -211,6 +237,39 @@ func StartSyncer(ctx context.Context, cfg *SyncerConfig, numSyncerThreads int, i
 		return err
 	}
 
+	ddsifForUpstreamSyncer, err := ddsif.NewDiscoveringDynamicSharedInformerFactory(upstreamSyncerClusterClient, nil, nil, resourceController, cache.Indexers{})
+	if err != nil {
+		return err
+	}
+
+	ddsifForUpstreamUpsyncer, err := ddsif.NewDiscoveringDynamicSharedInformerFactory(upstreamUpsyncerClusterClient, nil, nil,
```

Comment:
am confused. Multiple ddsifs for upstream? Why?

Reply:
There are 2 distinct upstream virtual workspaces: one for the Syncer, one for the UpSyncer.
Since the VW URLs are distinct, so are the kube clients and finally the ddsifs.